plugins/ptp_operator/metrics/metrics.go

```diff
@@ -0,0 +1,381 @@
+
+// RegisterMetrics ... register metrics for all side car plugins
+func RegisterMetrics(nodeName string) {
+	registerMetrics.Do(func() {
+		prometheus.MustRegister(OffsetFromMaster)
+		prometheus.MustRegister(MaxOffsetFromMaster)
+		prometheus.MustRegister(FrequencyAdjustment)
+		prometheus.MustRegister(DelayFromMaster)
+
+		// Including these stats kills performance when Prometheus polls with multiple targets
+		prometheus.Unregister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+		prometheus.Unregister(prometheus.NewGoCollector())
+
+		NodeName = nodeName
+	})
+}
+
+// UpdatePTPMetrics ...
+func UpdatePTPMetrics(process, iface string, offsetFromMaster, maxOffsetFromMaster, frequencyAdjustment, delayFromMaster float64) {
+	OffsetFromMaster.With(prometheus.Labels{
+		"process": process, "node": NodeName, "iface": iface}).Set(offsetFromMaster)
+
+	MaxOffsetFromMaster.With(prometheus.Labels{
+		"process": process, "node": NodeName, "iface": iface}).Set(maxOffsetFromMaster)
+
+	FrequencyAdjustment.With(prometheus.Labels{
+		"process": process, "node": NodeName, "iface": iface}).Set(frequencyAdjustment)
+
+	DelayFromMaster.With(prometheus.Labels{
+		"process": process, "node": NodeName, "iface": iface}).Set(delayFromMaster)
+}
+
+func extractSummaryMetrics(processName, output string) (offsetFromMaster, maxOffsetFromMaster, frequencyAdjustment, delayFromMaster float64) {
+	// remove everything before the rms string
+	// This makes the out to equals
+	indx := strings.Index(output, "rms")
+	output = output[indx:]
+	fields := strings.Fields(output)
+
+	if len(fields) < 5 {
+		glog.Errorf("%s failed to parse output %s: unexpected number of fields", processName, output)
+		return
+	}
+
+	offsetFromMaster, err := strconv.ParseFloat(fields[1], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse offset from master output %s error %v", processName, fields[1], err)
+	}
+
+	maxOffsetFromMaster, err = strconv.ParseFloat(fields[3], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse max offset from master output %s error %v", processName, fields[3], err)
+	}
+
+	frequencyAdjustment, err = strconv.ParseFloat(fields[5], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse frequency adjustment output %s error %v", processName, fields[5], err)
+	}
+
+	if len(fields) >= 10 {
+		delayFromMaster, err = strconv.ParseFloat(fields[9], 64)
+		if err != nil {
+			glog.Errorf("%s failed to parse delay from master output %s error %v", processName, fields[9], err)
+		}
+	} else {
+		// If there is no delay from master this mean we are out of sync
+		glog.Warningf("no delay from master process %s out of sync", processName)
+	}
+
+	return
+}
+
+func extractRegularMetrics(processName, output string) (offsetFromMaster, maxOffsetFromMaster, frequencyAdjustment, delayFromMaster float64, clockState ceevent.SyncState) {
+	// remove everything before the rms string
+	// This makes the out to equals
+	output = strings.Replace(output, "path", "", 1)
```

Comment:
why do you need to remove "path"?

Reply:
because ptp4l has only `delay`  and phc2sys has `path delay`, so it makes the output equal

Reply:
ok.